### ISSUE TITLE
Backdrop stays after dismissing

### ISSUE THREAD

**Issue by adhayward** [makeusabrew/bootbox#766]
On bootstrap 4.5.1 a bootbox alert with `backdrop: true` doesn't dismiss properly.
Clicking the OK or X (Close) buttons works fine but if you click on the backdrop, it hides the modal itself and leaves the backdrop displayed.

This isn't an issue when using bootstrap 4.5.0 so something must have been changed there to cause this however using a standard bootstrap modal dialog allows you to click the backdrop to dismiss and it works fine.

See [https://jsfiddle.net/fuzzyexponent/2cmxLywa/](https://jsfiddle.net/fuzzyexponent/2cmxLywa/) for example

**Comment by leoliu** [comment]
I ran into this bug as well after upgrading to bs 4.5.2. Have to revert to 4.5.0 as a workaround for now.

**Comment by tiesont** [comment]
Bootstrap 4.5.1 adds code to (apparently) correct overflowing modals (see https://github.com/twbs/bootstrap/blob/v4.5.1/js/src/modal.js#L241). That might be affecting our modals (if that code assumes the modal is pulled from page content, rather than the dynamic modal we generate). The only other change is a revamp in how they setup their object maps - I'd have to do some digging to see how they differ, which I can't do for a while, if at all.

**Comment by adhayward** [comment]
I think I've traced the issue to this line:
https://github.com/makeusabrew/bootbox/blob/9caa2d9048331e226e653e3c3c2e39e3f3eb2e7b/bootbox.js#L496

If I change it to `backdrop: options.backdrop,` it seems to work as expected for me. 

Is there a reason it's telling bootstrap the modal had a static backdrop even when I set bootbox to something different?

**Comment by adhayward** [comment]
I've added [PR #767](https://github.com/makeusabrew/bootbox/pull/767) that changes that line along with a bit of code in sanitize that forces the backdrop option to be `true`/`false`/`'static'` so that it works with anything passed in. I've made `false` or `0` handled as `false`, any other falsy values (e.g. `null`, `undefined`) or the string `'static'` (in any case) handled as `'static'` and any other non-falsy value becomes `true`.

I've also made is so the backdrop click handler only gets applied for `backdrop === true` rather than `backdrop !== 'static'` to match the behaviour of the default bootstrap modal (you could previously close the modal with backdrop=false by clicking the invisible backdrop)

**Comment by tiesont** [comment]
Thank you looking into this. I'll try to carve out some time today or tomorrow to review this.

**Comment by tiesont** [comment]
Okay, this breaks at least two of the tests. I'll need to dig into whether this indicates a problem, or if the tests are just built on incorrect/old assumptions.

**Comment by tiesont** [comment]
> I've also made is so the backdrop click handler only gets applied for backdrop === true rather than backdrop !== 'static' to match the behaviour of the default bootstrap modal (you could previously close the modal with backdrop=false by clicking the invisible backdrop)

@tarlepp This _is_ a behavior change (hence why two of the unit tests fail) - does this work for you? I _think_ it actually makes Bootbox more consistently follow what we claim on http://bootboxjs.com/documentation.html#bb-option-backdrop

**Comment by tarlepp** [comment]
@tiesont might be, I was just reading that #767 code - didn't test that.

**Comment by tiesont** [comment]
@tarlepp I ran the tests, and updating the broken tests in bootbox.test.js to this, everything passes:

```js
    describe('when set to false', function() {
      beforeEach(function() {
        this.callback = sinon.spy();
        this.dialog = bootbox.alert({
          message: 'hi',
          callback: this.callback,
          backdrop: false
        });
      });

      describe('When triggering the backdrop click dismiss event', function() {
        describe('With the wrong target', function() {
          beforeEach(function() {
            this.e({an: 'object'});
          });

          it('does not invoke the callback', function() {
            expect(this.callback).not.to.have.been.called;
          });
        });

        describe('With the correct target', function() {
          beforeEach(function() {
            this.e(this.dialog.get(0));
          });

          it('invokes the callback', function() {
            expect(this.callback).not.to.have.been.called;
          });
        });
      });
    });
```
It was failing before because the test expects the callback to be called when `backdrop: false` is used (since that's how the current code works). With the change above, everything passes.

I think I'm okay with this change, although I would need to step through some "live human" tests to make sure it works like I expect.

**Comment by tiesont** [comment]
Closed in #767
